File: facke-iot-service/src/facke-iot/domain/machine.dto.ts
import { StoppingPointDto } from "./stopping-point.dto";
import { ProductionDto } from "./production.dto";

export class MachineDto {

  private id: number;

  private name: string;

  private productions: ProductionDto[] = [];

  private stoppingPoints: StoppingPointDto[] = [];

  private piecesPerMinute: number;

  private maximumProductionCapacity: number;

  private workingTime: number;

  constructor(machineDto: MachineDto) {
    this.id = machineDto.id;
    this.name = machineDto.name;
    this.productions = machineDto.productions;
    this.stoppingPoints = machineDto.stoppingPoints;
    this.piecesPerMinute = machineDto.piecesPerMinute;
    this.maximumProductionCapacity = machineDto.maximumProductionCapacity;
    this.workingTime = machineDto.workingTime;
  }
}

File: iot-hub-api/src/main/java/com/soft/iot_hub_api/dto/MachineDTO.java
package com.soft.iot_hub_api.dto;

import com.soft.iot_hub_api.domain.Machine;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.beans.BeanUtils;

import java.io.Serializable;
import java.util.HashSet;
import java.util.Set;

@Data
@AllArgsConstructor
@NoArgsConstructor
@Builder
public class MachineDTO implements Serializable {

    private Long id;

    private String name;

    private Set<ProductionDTO> productions = new HashSet<>();

    private Set<StoppingPointDTO> stoppingPoints = new HashSet<>();

    private Integer piecesPerMinute;

    private Integer maximumProductionCapacity;

    private Integer workingTime;

    public MachineDTO(Machine machine) {
        BeanUtils.copyProperties(machine, this);
    }
}
